packages/driver/src/cy/commands/actions/attachFile.ts

```diff
@@ -6,23 +6,111 @@ import $errUtils from '../../../cypress/error_utils'
 import $actionability from '../../actionability'
 import { addEventCoords, dispatch } from './trigger'
 
-const createFileList = (files: Cypress.FileReferenceObject[]): FileList => {
+/* dropzone.js relies on an experimental, nonstandard API, webkitGetAsEntry().
+ * https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem/webkitGetAsEntry
+ * The behavior here varies between browsers, and though unpleasant, we have to attempt
+ * to replicate it as far as possible.
+ *
+ * In Firefox, attempting to set webkitGetAsEntry() fails - but this is fine, because
+ * Firefox also returns a useful value from item.webkitGetAsEntry().
+ *
+ * In webkit browsers however, item.webkitGetAsEntry() returns null, because our File()
+ * instances don't actually correspond with a file on disk. But the property is writeable,
+ * so we're able to override it with our own implementation that returns a proper object.
+ */
+const tryMockWebkit = (item) => {
+  try {
+    item.webkitGetAsEntry = () => {
+      return {
+        isFile: true,
+        file: (callback) => callback(item.getAsFile()),
+      }
+    }
+  } catch (e) {
+    // We're in Firefox, this is fine.
+  }
+
+  return item
+}
+
+const createDataTransfer = (files: Cypress.FileReferenceObject[]): DataTransfer => {
   const dataTransfer = new DataTransfer()
 
   files.forEach(({ contents, fileName, lastModified, mimeType }) => {
-    const file = new File([contents], fileName || '', { lastModified, type: mimeType })
+    dataTransfer.items.add(new File([contents], fileName || '', {
+      lastModified,
+      type: mimeType,
```

Comment:
you added a fallback value for fileName. Is there a chance lastModified or mimeType will be undefined? or does it matter to pass with these values?

Reply:
I originally added this to mollify typescript, but then trying to justify it in a comment here, I realized that the handling of default values was weird. Reworked in an incoming commit.

Also, mimeType is not needed at all, a vestige of cypress-file-upload that I hadn't completely discarded, so finished removing it.